ddtrace/appsec/_iast/_ast/ast_patching.py

```diff
@@ -427,30 +456,34 @@ def astpatch_module(module: ModuleType, remove_flask_run: bool = False) -> Tuple
     if module_ext.lower() not in {".pyo", ".pyc", ".pyw", ".py"}:
         # Probably native or built-in module
         log.debug("extension not supported: %s for: %s", module_ext, module_path)
-        return "", ""
+        return "", None
 
     with open(module_path, "r", encoding=get_encoding(module_path)) as source_file:
         try:
             source_text = source_file.read()
         except UnicodeDecodeError:
             log.debug("unicode decode error for file: %s", module_path, exc_info=True)
-            return "", ""
+            return "", None
 
     if len(source_text.strip()) == 0:
         # Don't patch empty files like __init__.py
         log.debug("empty file: %s", module_path)
-        return "", ""
+        return "", None
 
     if remove_flask_run:
         source_text = _remove_flask_run(source_text)
 
-    new_source = visit_ast(
+    if not asbool(os.environ.get(IAST.ENV_NO_DIR_PATCH, "false")):
+        # Add the dir filter so __ddtrace stuff is not returned by dir(module)
+        source_text += _DIR_WRAPPER
```

Comment:
I believe we should do this after patching the code, and only if the code was patched, which is done in the next line, right?

Reply:
I mean, this change should go in the AST visitor, just like we introduce the new imports of the aspects if required, we should add this to the end

Reply:
Yes. However, it's not easy to add it after patching because it's not as easy as adding the source code (with is already compiled to AST), but I will take a look at how to do it.